File: math/basic/Greatest_digit_in_number/javascript/greatest_digit_in_number.js
function greatest_digit_in_number(num) {
	num = num.toString();
	var a = num.split("");
	var max = a[0];
	a.forEach(element => {
		if(max < element) {
			max = element;
		}
	});
	return max;
}

// Example
var number = 37847862;
var digit = greatest_digit_in_number(number);
console.log(digit); // Expected output: 8
// Another example
const greatestDigitInNumber = n => n.toString().split('').reduce((max, current) => max = max < current ? current : max , 0);

File: math/basic/Greatest_digit_in_number/c++/GreatestDigitInNumber.cpp
#include<iostream>
#include<string>

using namespace std;
char Greatest_digit_in_number(string number);
int main(){
   string number;
   cin>>number;
   cout<<"The Greatest digit in this number is: "<<Greatest_digit_in_number(number)<<endl;
return 0;
}

char Greatest_digit_in_number(string number){
   int i =0;
   char greatest_digit = '0';
   for (char digit : number){
      if(digit > greatest_digit)
         greatest_digit = digit;
   }
   return greatest_digit;      
}
